fw/http.c

```diff
@@ -5346,38 +5355,38 @@ tfw_h2_resp_adjust_fwd(TfwHttpResp *resp)
 	 * processed above and which have non-empty value (i.e. configured
 	 * not for deletion).
 	 */
-	r = tfw_http_sess_resp_process(resp, false);
+	r = tfw_http_sess_resp_process(resp, stream_id, false);
```

Comment:
Now we pass `stream_id` to lot of functions, it looks redundant because each of these functions has `resp` argument which has `stream`. Can we just verify state of stream in beginning of `tfw_h2_resp_adjust_fwd` and then just use something like `tfw_h2_stream_id` but without spinlock? A as I see spinlock needs only to prevent races between stream deleting and reading, if so we can move streams to removing only after message transformation.

Reply:
You are not right here. Response stream usually is invalid field. It contains some necessary fields like `body_start_data` but for HTTP1.1 but it's stream id is 0. It's just a pointer to conn.stream. But i think we can implement additional field in resp structure, like req->stream and use it or move streams to half closed state later